src/lightning_app/cli/lightning_cli.py

```diff
@@ -141,6 +145,93 @@ def logs(app_name: str, components: List[str], follow: bool) -> None:
         rich.print(f"[{color}]{log_event.component_name}[/{color}] {date} {log_event.message}")
 
 
+@show.group()
+def cluster():
+    """Groups cluster commands inside show."""
+    pass
+
+
+@cluster.command(name="logs")
+@click.argument("cluster_name", required=True)
+@click.option(
+    "--from",
+    "from_time",
+    default="24 hours ago",
+    help="The starting timestamp to query cluster logs from.",
+    callback=_arrow_time_callback,
+)
+@click.option(
+    "--to",
+    "to_time",
+    default="0 seconds ago",
+    callback=_arrow_time_callback,
+    help="The end timestamp / relative time increment to query logs for. This is ignored when following logs (with "
+    "-f/--follow).",
+)
+@click.option("--limit", default=1000, help="The max number of log lines returned.")
+@click.option("-f", "--follow", required=False, is_flag=True, help="Wait for new logs, to exit use CTRL+C.")
+def cluster_logs(cluster_name: str, to_time: arrow.Arrow, from_time: arrow.Arrow, limit: int, follow: bool) -> None:
+    """Show cluster logs.
+
+    Example uses:
+
+        Print cluster logs:
+
+            $ lightning show cluster logs my-cluster
```

Comment:
Do you think we could converge to

```
lightning show [what] [name or id]
```

in the future? This was our vision initially. The id or name of the resource would identify whether it is an app, cluster, whatever (from looking up in database). So instead of

```
lightning show cluster logs my-cluster
```

we would have

```
lightning show logs my-cluster
```

but you could also have

```
lightning show logs my-app
```

The arguments --from and --to generalize to all of them too.

Reply:
Agree with @awaelchli - If possible we should show logs by resource id. So one `show logs <res>` would exists, instead of separate "per-resource-type".
I could help out with that functionality if needed, let me know ;)

Reply:
I think this can be a good next step, sure! For this stage, `lightning show cluster logs my-cluster` is the approved option, so I think it will be good to leave it untouched here.

Reply:
(Probably we should raise this discussion to a product level in the future)

Reply:
The concern is this will "never" get addressed and we'll end up having an inconsistent CLI

Reply:
TBH, I'm not a fan of the idea where we need to guess what type the `resource_name` in `lightning show logs <resource_name>` has, it can be confusing for users and for our logic too (although we can define type in almost all cases, we can face confusions e.g. if we have app and cluster named `test`, and should restrict the same names for resources). Maybe we should also consider something like `lightning show logs app <app_name>` or `lightning show logs cluster <cluster_name>`.